python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -374,14 +375,29 @@ def _replicate(
 
             end_time = time.time()
 
+            has_asset_def: bool = bool(context and context.has_assets_def)
+
             # TODO: In the future, it'd be nice to yield these materializations as they come in
             # rather than waiting until the end of the replication
             for stream in stream_definition:
-                output_name = dagster_sling_translator.get_asset_key(stream)
-                yield AssetMaterialization(
-                    asset_key=output_name,
-                    metadata={"elapsed_time": end_time - start_time, "stream_name": stream["name"]},
-                )
+                asset_key = dagster_sling_translator.get_asset_key(stream)
+
+                if has_asset_def:
+                    yield MaterializeResult(
+                        asset_key=asset_key,
+                        metadata={
```

Comment:
Rather than repeating the metadata dict in this block and the one below, it would be better to construct it just once and pass it in both palces

Reply:
Good calls - I've addressed both items. Thank you.